### PR TITLE
Implement Uuid codec for heed

### LINKED ISSUE #2984: Implement a UuidV4 codec for heed
Instead of [converting the uuidv4 to string before putting them in heed](https://github.com/meilisearch/meilisearch/blob/main/index-scheduler/src/index_mapper.rs#L33) we could create a custom codec.

Here you can see a list of heed codec to help you design the one for the Uuidv4: https://github.com/meilisearch/milli/tree/main/milli/src/heed_codec

### PR DESCRIPTION
**Pull request by elbertronnie**
# Pull Request

## Related issue
Fixes #2984 

## What does this PR do?
- Created a new heed codec for uuid::Uuid named as UuidCodec
- Replaced SerdeBincode\<Uuid\> with UuidCodec
- Removed the TODO in code associated with this issue

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: index-scheduler/src/index_mapper.rs [hunk 1]
```diff
@@ -228,3 +231,22 @@ impl IndexMapper {
...
+    }
+}
+
+impl BytesEncode<'_> for UuidCodec {
+    type EItem = Uuid;
+
+    fn bytes_encode(item: &Self::EItem) -> Option<Cow<[u8]>> {
+        Some(Cow::Borrowed(item.as_bytes()))
+    }
+}
```
Comment (Kerollmops) [inline index-scheduler/src/index_mapper.rs:252]: Would it be possible to move this impl to another distinct file, named `uuid_codec`, maybe?

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you for your PR, looks quite good!

File: index-scheduler/src/index_mapper.rs (hunk 1 above)
Comment (elbertronnie) [reply index-scheduler/src/index_mapper.rs]: I have moved this to a new file named uuid_codec.rs.
I actually forgot to ask where should this file be situated? Currently it is in the same folder as `index_mapper.rs`. Is this fine or should it moved to different folder?

Comment (Kerollmops) [reply index-scheduler/src/index_mapper.rs]: That's perfect, thank you!

**Review by Kerollmops — APPROVED** [review]
Thank you for your contrib'!
bors merge

**Comment by curquiza** [comment]
bors merge

**Comment by Kerollmops** [comment]
bors merge

**Comment by Kerollmops** [comment]
bors abort

**Comment by Kerollmops** [comment]
bors cancel

**Comment by curquiza** [comment]
bors merge-

**Comment by Kerollmops** [comment]
bors merge

**Pull request merged**